### PR TITLE
Threads database, resolvers, and sample query (part 1)

### PR DESCRIPTION
**Pull request by jerrylee17**
- Schema set up
- Resolvers have crud functionality
- Sample query has been set up

Up next:
- Change to dotEnv
- Customize resolver to modify threadMessage object inside thread object
- Make the query functions easier to use
- Connect the API functions with threads command

### CODE REVIEW

**Comment by hsorathia** [comment]
some of the queries you gave me:
create a thread messages:
```
mutation {
  threadMessageCreateOne (
    record:{
      messageID:"4123",
      position:1
    }
  ) {
    recordId
    record{
      messageID
      position
    }
  }
}
```
create a thread: 
```
# Write your query or mutation here
mutation {
  threadCreateOne(
    record:{
      threadID: "100000",
      topic: "bigbrain",
      creatorID: "5000",
      guildID:"3000",
      messages: [
        "threadmessageID"
      ]
    }
  ) {
    recordId
    record {
      threadID
      creatorID
      guildID
      topic
      threadMessages
    }
  }
}
```

query to see a thread:
```
query {
  threadMany {
    threadID
    creatorID
    guildID
    topic
    messages {
      messageID
      position
    }
  }
}
```

see all the thread messages: 
```
query {
  threadMessageMany {
    _id
    messageID
    position
  }
}
```

File: api/models/thread.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  { collection: 'threads' }
+);
+
+const Thread = mongoose.model('Thread', ThreadSchema);
+const ThreadTC = composeMongoose(Thread);
+
+// Add relations
+ThreadTC.addRelation(
+  'threadMessages',
```
Comment (hsorathia) [inline api/models/thread.js:40]: good fix

File: api/models/thread.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const mongoose = require('mongoose');
+const { composeMongoose } = require('graphql-compose-mongoose');
+const { ThreadMessageTC } = require('./threadMessage');
+
+
+const ThreadSchema = mongoose.Schema(
+  {
+    threadID: {
+      // Automatically assign using timestamp
```
Comment (hsorathia) [inline api/models/thread.js:9]: what does this mean

File: src/APIFunctions/thread.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const { request, gql } = require('graphql-request');
+const { url } = require('../../config.json');
```
Comment (hsorathia) [inline src/APIFunctions/thread.js:2]: consider changing to dotenv()
i'm pretty sure that's industry standard:tm: but it's up to you. this isn't much different

**Review by hsorathia — APPROVED** [review]
works for me :+1:

nice work!

File: api/models/thread.js (hunk 2 above)
Comment (jerrylee17) [reply api/models/thread.js:9]: Generate the threadID from the client side

File: src/APIFunctions/thread.js (hunk 3 above)
Comment (jerrylee17) [reply src/APIFunctions/thread.js:2]: Next patch

**Pull request merged**